cli/cppcheckexecutor.cpp

```diff
@@ -66,11 +66,26 @@
 #include <windows.h>
 #endif
 
+class XMLErrorMessagesLogger : public ErrorLogger
+{
+    void reportOut(const std::string & outmsg, Color /*c*/ = Color::Reset) override
+    {
+        std::cout << outmsg << std::endl;
+    }
+
+    void reportErr(const ErrorMessage &msg) override
+    {
+        reportOut(msg.toXML());
+    }
+
+    void reportProgress(const std::string & /*filename*/, const char /*stage*/[], const std::size_t /*value*/) override
```

Comment:
@danmar 
I think we can remove `reportProgress()`. It is supposed to give progress messages at certain (albeit very few) stages which might be long-running. But actually getting such a message looks very unlikely and it doesn't really provide much value. It is not helpful to the user since there is nothing he can do about it. And if we get a report about a slow analysis this we simply profile it since we need the actual cause not just not some ballpark string.

I do see it being used as an cancellation callback but as that callback is hit very rarely that will probably not work as expected. That could be implemented via the `Settings::terminated()` check instead which is performed more often.

Sorry for having yet another off-topic discussion in a PR (we should really enable the discussions on the repo).

Reply:
The idea was that if analysis of a file takes minutes.. it will be helpful for the user to see that there is progress and that we haven't hanged. It was mostly for the GUI. Unfortunately it's not used in the GUI as far as I know it's not trivial to add it properly (technically it's easy but ui-wise not so easy). Technically we could add labels and progress bars for each thread in the mainwindow but it will waste precious space...

Reply:
In the command line a message is written every ~ 10 seconds. But the idea was to update the progress in the GUI more often maybe every seconds or so..

Reply:
See #5269.